src/python/pants/jvm/target_types.py

```diff
@@ -216,3 +218,22 @@ def validate(self) -> None:
 
 class JunitTestSourceField(SingleSourceField, metaclass=ABCMeta):
     """A marker that indicates that a source field represents a JUnit test."""
+
+
+# -----------------------------------------------------------------------------------------------
+# Jvm Resources Fields
+# -----------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class JvmResourcesFieldSet(FieldSet):
+    required_fields = (ResourceSourceField,)
+
+    sources: ResourceSourceField
+
+
+@dataclass(frozen=True)
+class JvmResourcesGeneratorFieldSet(FieldSet):
+    required_fields = (ResourcesGeneratingSourcesField,)
+
+    sources: ResourcesGeneratingSourcesField
```

Comment:
I don't think this is right. Pants will already run on the generated `resource` targets, no need to run on the generator too.

Reply:
Thanks! I wasn't sure about this, so just went with what Stu suggested in the original ticket. I'll make the change

Reply:
@Eric-Arellano It looks like the resources generator is necessary, or at the very least, things stop working if you have a `resources(name="foo")` and a `jvm_sources(dependencies=[":foo"])` type arrangement. I'll keep it in there for the time being.

Reply:
note that this matches practice elsewhere in the codebase (e.g. for [Java compiles](https://github.com/pantsbuild/pants/blob/b8c114f1c990b8e86dbda0faf42985f79102dc33/src/python/pants/backend/java/compile/javac.py#L36-L37))

Reply:
Specifically in tests, or in the wild too? If in tests, you probably need to activate core.target_types.rules

Reply:
This was during a smoke test using the JVM testproject. No JUnit tests were involved.

Reply:
Hm okay. Thanks for linking to the other stuff. This is fine for now, I'll try to figure out why expansion isn't working.